eventlogger/src/main/java/gov/cms/ab2d/eventlogger/eventloggers/sql/ApiRequestEventMapper.java

```diff
@@ -3,12 +3,64 @@
 import gov.cms.ab2d.eventlogger.EventLoggingException;
 import gov.cms.ab2d.eventlogger.LoggableEvent;
 import gov.cms.ab2d.eventlogger.events.ApiRequestEvent;
+import gov.cms.ab2d.eventlogger.utils.UtilMethods;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.OffsetDateTime;
+
+public class ApiRequestEventMapper extends SqlEventMapper {
+    private JdbcTemplate template;
+
+    public ApiRequestEventMapper(JdbcTemplate template) {
+        this.template = template;
+    }
 
-public class ApiRequestEventMapper implements SqlEventMapper {
     @Override
     public void log(LoggableEvent event) {
         if (event.getClass() != ApiRequestEvent.class) {
             throw new EventLoggingException("Used " + event.getClass().toString() + " instead of " + ApiRequestEvent.class.toString());
         }
+        String query = "insert into event_api_request " +
+                " (time_of_event, user_id, job_id, url, ip_address, token_hash, request_id) " +
+                " values (?, ?, ?, ?, ?, ?, ?)";
+
+        ApiRequestEvent be = (ApiRequestEvent) event;
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        template.update(connection -> {
+            PreparedStatement ps = connection
+                    .prepareStatement(query, new String[] {"id"});
+            ps.setObject(1, UtilMethods.convertToUtc(be.getTimeOfEvent()));
+            ps.setString(2, be.getUser());
+            ps.setString(3, be.getJobId());
+            ps.setString(4, be.getUrl());
+            ps.setString(5, be.getIpAddress());
+            ps.setString(6, be.getTokenHash());
+            ps.setString(7, be.getRequestId());
+            return ps;
+        }, keyHolder);
```

Comment:
Is this not using hibernate since it's out of the application's context?

Reply:
Because I'm separating the event from the logging (so we can log the same event to different places), the event isn't a management entity.